src/types.rs

```diff
@@ -20,34 +22,193 @@ pub struct ModeChangeError<CommE, PinE, DEV> {
 
 /// Device operation modes
 pub mod mode {
-    /// Magnetometer one-shot (single) mode
+    /// Marker type for magnetometer in one-shot (single) mode.
     #[derive(Debug)]
-    pub struct MagOneShot;
-    /// Magnetometer continuous mode
+    pub enum MagOneShot {}
```

Comment:
Any reason for this change? (also below)

Reply:
These are only used as markers, so they never need to be constructed. This just makes this fact part of the type.

Reply:
Ah, I get it now. Thanks!